### PR TITLE
chore(wallet): avoid chain/account switch in Swap

### PR DESCRIPTION
**Pull request by onyb**
Resolves https://github.com/brave/brave-browser/issues/34224

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

Swaps in desktop and android should continue to work. Make sure you try switching network and account in the respective dropdowns, and the balances should be fetched correctly.

### CODE REVIEW

File: components/brave_wallet_ui/page/screens/swap/components/swap/header/header.tsx [hunk 1]
```diff
@@ -142,7 +134,7 @@ export const Header = () => {
             hasBackground={true}
             hasShadow={true}
             isHeader={true}
-            networkNotSupported={!isNetworkSupported}
+            networkNotSupported={false}
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/page/screens/swap/components/swap/header/header.tsx:137]: Do we need this prop anymore?

Comment (onyb) [reply components/brave_wallet_ui/page/screens/swap/components/swap/header/header.tsx:137]: Nope. If you're okay with me removing the associated React code, I can get rid of the prop.

**Review by josheleonard — APPROVED** [review]

File: components/brave_wallet_ui/page/screens/swap/components/swap/header/header.tsx (hunk 1 above)
Comment (onyb) [reply components/brave_wallet_ui/page/screens/swap/components/swap/header/header.tsx]: removed here: https://github.com/brave/brave-core/pull/20923/commits/a8fea1d18cfc48a7ba897f97414d7c93f2b10aa9

**Review by yrliou — APPROVED** [review]
core++

**Review by mkarolin — APPROVED** [review]
`strings`++

File: components/brave_wallet_ui/page/screens/swap/hooks/useSwap.ts [hunk 2]
```diff
@@ -628,27 +689,24 @@ export const useSwap = () => {
...
-    selectedNetwork,
-    getLocale
-  ])
+  }, [fromToken, swapValidationError, selectedNetwork, getLocale])
 
   const isSubmitButtonDisabled = useMemo(() => {
     return (
+      !selectedNetwork ||
+      !selectedAccount ||
+      selectedNetwork.coin !== selectedAccount.accountId.coin ||
```
Comment (supermassive) [inline components/brave_wallet_ui/page/screens/swap/hooks/useSwap.ts:698]: That works for ETH and SOL, but doesn't in general. So for consistency if we need to match account with network let's use `networkSupportsAccount` from network-utils.ts which also works for other coins where accounts for different networks(testnet/mainnet) are incompatible.
@onyb

Comment (onyb) [reply components/brave_wallet_ui/page/screens/swap/hooks/useSwap.ts:698]: Nice catch! I'll fix it in a subsequent PR.

**Pull request merged**